lib/graphql.js

```diff
@@ -0,0 +1,131 @@
+/** Copyright (c) 2019 Uber Technologies, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import get from 'just-safe-get';
+import {graphql} from './github.js';
+
+/**
+ * @typedef {{
+ *   repoName: string,
+ * }} BasePropsType
+ */
+
+/**
+ * @param {query} string
+ * @param {object} props
+ * @param {string} [path]
+ * @returns {Promise<*>}
+ */
+async function query(query, props, path) {
+  // replace `repoName` prop with `owner` and `repo` for convenience
+  if (props.repoName) {
+    props = {...props};
+    const [owner, repo] = props.repoName.split('/');
+
+    delete props.repoName;
+    Object.assign(props, {owner, repo});
+  }
+
+  return (
+    // inject rate limit fields
+    graphql(
+      query.replace(/\}$/, ' rateLimit {limit, remaining}, viewer {login}}'),
+      props,
+    )
+      .then(res => {
+        const {rateLimit, viewer} = res || {};
+
+        if (viewer && rateLimit && rateLimit.remaining % 10 === 0) {
+          console.log(
+            `${viewer.login}: remaining graphql points: ${rateLimit.remaining}`,
+          );
+        }
+
+        return res;
+      })
+      .then(res => (path ? get(res, path) : res))
+  );
+}
+
+/**
+ * @param {string} fields fields to provide to the query
+ * @param {BasePropsType & {
+ *   branchName: string,
+ *   includeClosed?: boolean,
+ * }} props
+ * @param {string} [path] dot-notation path to return from response
+ * @returns {Promise<*>}
+ */
+export function getPRsFromBranch(fields, props, path) {
+  // `includeClosed` should only be used when 100% positive there's an active
+  // partner pr with the provided branch name (i.e. when the partner pr number
+  // is in the branch name), otherwise results will likely include old, already-
+  // merged branches
+  const states = props.includeClosed ? 'CLOSED, MERGED, OPEN' : 'OPEN';
+
+  props = {...props};
+  delete props.includeClosed;
```

Comment:
You could alternatively do something like:

```js
const {includeClosed, ...filteredProps} = props;
```

Reply:
Good idea